net/adoptopenjdk/bumblebench/collections/ArrayCopyBench.java

```diff
@@ -0,0 +1,129 @@
+* limitations under the License.
+*******************************************************************************/
+
+package net.adoptopenjdk.bumblebench.collections;
+
+import net.adoptopenjdk.bumblebench.core.MicroBench;
+
+/**
+ * 
+ * 
+ * This benchmark simply tests the speed of copying arrays in either backward
+ * or forward direction. This direction can be specified by options.
+ * 
+ * The reported score is in terms of number of array copies done in one second.
+ * Length of array is fixed to 8192 but type can be changed using options.
+ * Also if not specified this benchmark will always test array copy for following types.
+ * rimitive Arrays(int, long, byte, character) and Object Arrays. This can be limited to
+ * specific type using options.
+ * 
+ * Also this benchmark in all iterations copies 4000 elements which can be changed
+ * through options.
+ * 
+ * */
+
+
+public final class ArrayCopyBench extends MicroBench {
+	private static final boolean BACKWARD_ARRAYCOPY = option("backwardArrayCopy", false);
+	private static final int ARRAY_LENGTH = option("arrayLength", 8192);
+	private static final int COPY_ELEMENTS = option("copyElements", 4000);
+	private static final int MAX_ITERATIONS_PER_LOOP = option("maxIterations", 10000000);
+	private static final boolean TESTINT = option("testInt", false);
+	private static final boolean TESTLONG = option("testLong", false);
+	private static final boolean TESTOBJ = option("testObj", false);
+	private static final boolean TESTCHAR = option("testChar", false);
+	private static final boolean TESTBYTE = option("testByte", false);
+
+	private static int[] intArray = new int[ARRAY_LENGTH];
+	private static char[] charArray = new char[ARRAY_LENGTH];
+	private static long[] longArray = new long[ARRAY_LENGTH];
+	private static byte[] byteArray = new byte[ARRAY_LENGTH];
+	private static String[] objArray = new String[ARRAY_LENGTH];
+	private static int dstPoint;
+	private static int srcPoint;
+	
+	static {
+		for (int i=0; i < ARRAY_LENGTH; i++) {
+			intArray[i] = i;
+			charArray[i] = (char)i;
+			longArray[i] = (long)i;
+			byteArray[i] = (byte)i;
+			objArray[i] = Integer.toString(i);
+		}
+
+		if (BACKWARD_ARRAYCOPY) {
+			srcPoint = 0;
+			dstPoint = 10;
+		}
+		else {
+			srcPoint=10;
+			dstPoint=0;
+		}
+	}
+
+	protected int maxIterationsPerLoop() {
+		return MAX_ITERATIONS_PER_LOOP;
+	}	
+
+	@Override
+	protected long doBatch(long numIterations) throws InterruptedException {
+		if (TESTINT) {	
+			for (long loop = 0; loop < numIterations; loop++) {
+				startTimer();
+				System.arraycopy(intArray, srcPoint, intArray, dstPoint, COPY_ELEMENTS);
+				pauseTimer();
+			}
+		}
+		else if (TESTLONG) {
+			for (long loop = 0; loop < numIterations; loop++) {
+				startTimer();
+				System.arraycopy(longArray, srcPoint, longArray, dstPoint, COPY_ELEMENTS);
+				pauseTimer();
+			}
+		}
+		else if (TESTOBJ) {
+			for (long loop = 0; loop < numIterations; loop++) {
+				startTimer();
```

Comment:
I don't think this is what we want. We want these timers outside of the `for` loops, correct? Also are `TESTLONG` and `TESTOBJ` supposed to be mutually exclusive? Can't someone request just those two to be tested? In which case your benchmark will only report the long test result. I'm concerned that we have `else if` statements here rather than just `if` IMO.

Reply:
Ahh, Understood what you meant to say. I added feature to allow more than single type of array copy test and if not specified it will run all type. 
Also I divided `doBatch` in separate workload here as it is not the method I am measuring perf of. In previous implementation as System.arraycopy would be inlined, compiler sees `doBatch` a hot method and tries to optimize it. 

As per my understanding start and pause timer should be around workload which is System.arrayCopy here.  (Looking at the description about microbench and minibench and some example -https://github.com/AdoptOpenJDK/bumblebench/blob/5774b0662fbee431e41ded4fd27d394cd59e3188/net/adoptopenjdk/bumblebench/collections/ArrayListRemoveIfLambdaBench.java#L22)